scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -226,62 +256,151 @@ private static void askInteractively(Console console, UserInfo userInfo, Action
+
+            if (sourceGroupFile != null) {
+                sourceGroupsMap = loadGroups(sourceGroupFile);
+            } else {
+                sourceGroupsMap = TreeMultimap.create();
+                if (userInfo != null && userInfo.getGroups() != null) {
+                    for (String group : userInfo.getGroups()) {
+                        sourceGroupsMap.put(userInfo.getLogin(), group);
                     }
-                    props.remove(userInfo.getLogin());
-                    groupsMap.removeAll(userInfo.getLogin());
-                    System.out.println("Deleted user " + userInfo.getLogin() + " in " + loginFilePath);
-                    break;
+                }
+            }
+            Collection<String> sourceLoginNames = sourceLoginProps.stringPropertyNames();
+            if (sourceLoginNames.isEmpty()) {
+                sourceLoginNames = sourceGroupsMap.keySet();
+            }
+
+            boolean bulkMode = sourceLoginNames.size() > 1;
+
+            for (String user : sourceLoginNames) {
+                UserInfo sourceUserInfo = new UserInfo();
+                sourceUserInfo.setLogin(user);
+                sourceUserInfo.setPassword((String) sourceLoginProps.get(user));
+                if (sourceGroupsMap.containsKey(user)) {
+                    sourceUserInfo.setGroups(sourceGroupsMap.get(user));
+                }
+
+                switch (action) {
+                    case CREATE:
+                        createAccount(pubKey, sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap, bulkMode);
+                        break;
+                    case UPDATE:
+                        updateAccount(pubKey, sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap, bulkMode);
+                        break;
+                    case DELETE:
+                        deleteAccount(sourceUserInfo, loginFilePath, groupFilePath, destinationLoginProps, destinationGroupsMap);
+                        break;
+                }
             }
 
-            storeLoginFile(loginFilePath, props);
+            storeLoginFile(loginFilePath, destinationLoginProps);
+
+            storeGroups(groupFilePath, destinationGroupsMap);
 
-            storeGroups(groupFilePath, groupsMap);
         } catch (Throwable t) {
             exitWithErrorMessage("Unexpected error", null, t);
         }
     }
 
+    private static void deleteAccount(UserInfo userInfo, String loginFilePath, String groupFilePath, Properties props, Multimap<String, String> groupsMap) throws ManageUsersException {
+        if (userInfo.getLogin().isEmpty()) {
+            warnWithMessage("Provided username is empty, skipping.");
+            return;
+        }
+        if (!props.containsKey(userInfo.getLogin())) {
+            warnWithMessage("USER " + userInfo.getLogin() + " does not exist in login file : " + loginFilePath);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "USER " 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal " does not exist in login file : " 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1192)